documents/error.go

```diff
@@ -1,122 +1,71 @@
+	ErrDocumentPersistence = errors.Error("error encountered when storing document in the system database")
 
-// Errors returns an array of errors
-func Errors(err error) []error {
-	if err == nil {
-		return nil
-	}
+	// ErrDocumentPackingCoreDocument must be used when packing of core document for the given document failed
+	ErrDocumentPackingCoreDocument = errors.Error("core document packing failed")
 
-	if multiErr, ok := err.(*multierror.Error); ok {
-		return multiErr.Errors
-	}
+	// ErrDocumentUnPackingCoreDocument must be used when unpacking of core document for the given document failed
+	ErrDocumentUnPackingCoreDocument = errors.Error("core document unpacking failed")
 
-	return []error{err}
-}
+	// ErrDocumentPrepareCoreDocument must be used when preparing a new core document fails for the given document
+	ErrDocumentPrepareCoreDocument = errors.Error("core document preparation failed")
 
-// LenError returns the amount of embedded errors.
-func LenError(err error) int {
+	// ErrDocumentSigning must be used when document signing related functionality fails
+	ErrDocumentSigning = errors.Error("document signing failed")
 
-	if err == nil {
-		return 0
-	}
+	// ErrDocumentAnchoring must be used when document anchoring fails
+	ErrDocumentAnchoring = errors.Error("document anchoring failed")
 
-	if multiErr, ok := err.(*multierror.Error); ok {
+	// ErrDocumentCollaborator must be used when there is an error in processing collaborators
+	ErrDocumentCollaborator = errors.Error("document collaborator issue")
 
-		return multiErr.Len()
-	}
-	return 1
+	// ErrDocumentProof must be used when document proof creation fails
+	ErrDocumentProof = errors.Error("document proof error")
+)
 
+// Error wraps an error with specific key
+// Deprecated: in favour of Error type in `github.com/centrifuge/go-centrifuge/errors`
+type Error struct {
+	key string
+	err error
 }
 
-func addToMap(errorMap map[string]string, key, msg string) map[string]string {
-	if errorMap[key] != "" {
-		errorMap[key] = fmt.Sprintf("%s\n%s", errorMap[key], msg)
-
-	} else {
-		errorMap[key] = msg
-
-	}
-	return errorMap
+// Error returns the underlying error message
+func (e Error) Error() string {
+	return fmt.Sprintf("%s : %s", e.key, e.err)
 }
 
-// ConvertToMap converts errors into a map.
-func ConvertToMap(err error) map[string]string {
-
-	errorMap := make(map[string]string)
-	var key string
-	var standardErrorCounter int
-
-	errors := Errors(err)
-
-	for _, err := range errors {
-		if err, ok := err.(Error); ok {
-			key = err.key
-
-		} else {
-			standardErrorCounter++
-			key = fmt.Sprintf("error_%v", standardErrorCounter)
-
-		}
-
-		addToMap(errorMap, key, err.Error())
-
-	}
-	return errorMap
-
+// NewError creates a new error from a key and a msg.
+// Deprecated: in favour of Error type in `github.com/centrifuge/go-centrifuge/errors`
+func NewError(key, msg string) error {
```

Comment:
Lets delete this. We dont need this anymore

Reply:
It creates more changes in `coredocument` package that uses this, I don't want to handle it in this pr.